pkg/pluginsupplier/plugin_supplier.go

```diff
@@ -5,54 +5,33 @@
 package pluginsupplier
 
 import (
-	"os"
 	"slices"
-	"strconv"
 
 	"github.com/vmware-tanzu/tanzu-cli/pkg/catalog"
 	"github.com/vmware-tanzu/tanzu-cli/pkg/cli"
-	"github.com/vmware-tanzu/tanzu-cli/pkg/constants"
 	configlib "github.com/vmware-tanzu/tanzu-plugin-runtime/config"
 	configtypes "github.com/vmware-tanzu/tanzu-plugin-runtime/config/types"
 )
 
-// GetInstalledPlugins return the installed plugins( both standalone and server plugins )
+// GetInstalledPlugins return the installed plugins
 func GetInstalledPlugins() ([]cli.PluginInfo, error) {
-	plugins, err := GetInstalledServerPlugins()
-	if err != nil {
-		return nil, err
-	}
-	standalonePlugins, err := GetInstalledStandalonePlugins()
-	if err != nil {
-		return nil, err
-	}
-	plugins = append(plugins, standalonePlugins...)
+	// Migrate context-scoped plugins as standalone plugin if required
+	// TODO(anujc): Think on how to invoke this function just once after the newer version
+	// of the CLI gets installed as we just need to do this migration once
+	catalog.MigrateContextPluginsAsStandaloneIfNeeded()
```

Comment:
https://github.com/vmware-tanzu/tanzu-cli/pull/723 can be useful for this situation.